### ISSUE TITLE
New windows_subsystem attribute does not warn or error when erroneous parameter passed in

### ISSUE THREAD

**Issue by johnthagen** [rust-lang/rust#42621]
See Rust [forum post](https://users.rust-lang.org/t/solved-console-window-still-opens-when-using-windows-subsystem-windows/11318) for more details.

[Rust 1.18](https://blog.rust-lang.org/2017/06/08/Rust-1.18.html) landed the `windows_subsystem` attribute. The blog post lists the expected usage as:

```
#![windows_subsystem(windows)]
```

But on this program it has no affect:

```rust
#![windows_subsystem(windows)]

use std::{thread, time};

fn main() {
    let ten_millis = time::Duration::from_secs(3);
    thread::sleep(ten_millis);
}
```

Even more strangely, arbitrary input is silently ignored with no warning or error:

```rust
#![windows_subsystem(batman)]

use std::{thread, time};

fn main() {
    let ten_millis = time::Duration::from_secs(3);
    thread::sleep(ten_millis);
}
```

I believe that there should be some kind of error or warning here to help users more quickly catch these sorts of errors, or perhaps the whole attribute has some other flaw that wasn't caught?

### Configuration

```
active toolchain
----------------

stable-x86_64-pc-windows-msvc (default)
rustc 1.18.0 (03fc9d622 2017-06-06)
```

**Comment by johnthagen** [comment]
CC Blog post issue: https://github.com/rust-lang/blog.rust-lang.org/issues/178

**Comment by brson** [comment]
The potential inconsistencies in how incorrect attributes are handled has always worried me.

**Comment by nikomatsakis** [comment]
We discussed this in the @rust-lang/lang team meeting. It seems like we should reject invalid values (the second case). I'm not sure what's up with the first example but presumably it is also a bug that it has no affect.

**Comment by cuviper** [comment]
It's supposed to be `#![windows_subsystem = "windows"]`, and that form does reject invalid values.  I suppose the bug is that it accepts the parenthesized form at all.

**Comment by rkarp** [comment]
Even just `#![windows_subsystem]` is also allowed, which can be misleading.

**Comment by Mark-Simulacrum** [comment]
This has since been fixed.
